scripts/lsp_types.lua

```diff
@@ -18,24 +19,117 @@ local function tofile(fname, text)
   end
 end
 
-function M.gen(opt)
-  if vim.uv.fs_stat('./lsp.json') then
-    vim.fn.delete('./lsp.json')
+local function sort_by_method(tbl)
+  local single, client, textD, workspace, others = {}, {}, {}, {}, {}
+  for _, item in ipairs(tbl) do
+    local parts = vim.split(item.method, '/', { trimempty = true })
+    if #parts == 1 then
+      single[#single + 1] = item
+    elseif parts[1] == 'textDocument' then
+      textD[#textD + 1] = item
+    elseif parts[1] == 'client' then
+      client[#client + 1] = item
+    elseif parts[1] == 'workspace' then
+      workspace[#workspace + 1] = item
+    else
+      others[#others + 1] = item
+    end
+  end
+
+  local res = {}
+  for _, item in ipairs({ single, client, textD, workspace, others }) do
+    res = vim.list_extend(res, item)
   end
-  vim.fn.system({
-    'curl',
-    'https://raw.githubusercontent.com/microsoft/lsprotocol/' .. opt.ref .. '/generator/lsp.json',
-    '-o',
-    './lsp.json',
+  return res
+end
+
+local function read_json(opt)
+  local uri = 'https://raw.githubusercontent.com/microsoft/language-server-protocol/gh-pages/_specifications/lsp/'
+    .. opt.version
+    .. '/metaModel/metaModel.json'
+
+  local res = vim.system({ 'curl', uri, '-o', '-' }):wait()
+
+  if res.code ~= 0 then
+    io.write(res.stderr)
+    return
```

Comment:
This doesn't fail on 404. Always test failure modes.

Pushed an update that tries to handle this.